plugins/builds/index.js

```diff
@@ -761,9 +794,14 @@ const buildsPlugin = {
                         parentBuildId: current.build.id
                     };
 
-                    newBuild = await createInternalBuild(internalBuildConfig);
-                } else {
-                    // nextBuild is not build model, so fetch proper build
+                    try {
+                        newBuild = await createInternalBuild(internalBuildConfig);
+                    } catch (err) {
+                        nextBuild = await getNextBuild(nextBuildConfig);
+                    }
+                }
+
+                if (!newBuild && nextBuild) {
                     newBuild = await updateParentBuilds({
                         joinParentBuilds: parentBuilds,
                         nextBuild: await buildFactory.get(nextBuild.id),
```

Comment:
can you just pass nextBuild directly here?
```suggestion
                        nextBuild
```